src/firefly/java/edu/caltech/ipac/firefly/server/AppServerCommands.java

```diff
@@ -137,21 +139,37 @@ public String doCommand(SrvParam params) throws Exception {
 
 
     private static final String HELP_BASE_URL = "help.base.url";
+    private static final String OP_ROOT = "OP_";
+    private static final int OP_ROOT_LENGTH = OP_ROOT.length();
     private static String getAppOptions() {
-        Map appOpts;
         String def = "{}";
         String appOptStr = AppProperties.getProperty(JsonProperty.FIREFLY_OPTIONS, def);
         try{
-            appOpts = (Map) new JSONParser().parse(appOptStr);
-            // additional props as FIREFLY_OPTIONS
-            applyIfNotEmpty(AppProperties.getProperty(HELP_BASE_URL), v -> appOpts.put(HELP_BASE_URL, v));
-
-            return new JSONObject(appOpts).toJSONString();
-        }catch(ParseException pe){
+            JsonHelper jhelp= JsonHelper.parse(appOptStr);
+            for(Map.Entry<String,String> entry : AppProperties.getWithStartingMatch(OP_ROOT).entrySet()) {
+                setToJson(jhelp, entry.getKey(), entry.getValue());
+            }
+            applyIfNotEmpty(AppProperties.getProperty(HELP_BASE_URL), v -> jhelp.setValue(v,HELP_BASE_URL));
+            return jhelp.toJson();
+        } catch(IllegalArgumentException pe){
             Logger.getLogger().error(String.format("Failed parsing %s", appOptStr));
             return def;
         }
     }
 
+    private static void setToJson(JsonHelper jhelp, String key, String valStr) {
+        if (key.length()>OP_ROOT_LENGTH) {
+            Object val;
+            if (valStr.endsWith("D") && valStr.length()>1) {
+                double num= getDouble(valStr.substring(0, valStr.length()-1));
+                val= Double.isNaN(num) ? valStr : num;
+            }
+            else {
+                val= valStr;
+            }
```

Comment:
Would it make more sense to convert any value into a number, and otherwise treat it as a string? If you have a string of numbers, you can enclose them in double quotes. I've seen this approach used elsewhere as well.

Reply:
I guess, this approach gives more control though.